datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/MLModelType.java

```diff
@@ -0,0 +1,115 @@
+
+    private static final Set<String> FACET_FIELDS = ImmutableSet.of("origin", "platform");
+    private static final String DEFAULT_AUTO_COMPLETE_FIELD = "name";
+    private final MLModels _mlModelsClient;
+
+    public MLModelType(final MLModels mlModelsClient) {
+        _mlModelsClient = mlModelsClient;
+    }
+
+    @Override
+    public EntityType type() {
+        return EntityType.MLMODEL;
+    }
+
+    @Override
+    public Class<MLModel> objectClass() {
+        return MLModel.class;
+    }
+
+    @Override
+    public List<MLModel> batchLoad(final List<String> urns, final QueryContext context) throws Exception {
+        final List<MLModelUrn> mlModelUrns = urns.stream()
+            .map(MlModelUtils::getMLModelUrn)
+            .collect(Collectors.toList());
+
+        try {
+            final Map<MLModelUrn, com.linkedin.ml.MLModel> mlModelMap = _mlModelsClient.batchGet(mlModelUrns
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet()));
+
+            final List<com.linkedin.ml.MLModel> gmsResults = mlModelUrns.stream()
+                .map(modelUrn -> mlModelMap.getOrDefault(modelUrn, null)).collect(Collectors.toList());
+
+            return gmsResults.stream()
+                .map(gmsMlModel -> gmsMlModel == null ? null : MlModelMapper.map(gmsMlModel))
+                .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to batch load MLModels", e);
+        }
+    }
+
+    @Override
+    public SearchResults search(@Nonnull String query,
+        @Nullable List<FacetFilterInput> filters,
+        int start,
+        int count,
+        @Nonnull final QueryContext context) throws Exception {
+        final Map<String, String> facetFilters = ResolverUtils.buildFacetFilters(filters, FACET_FIELDS);
+        final CollectionResponse<com.linkedin.ml.MLModel> searchResult = _mlModelsClient.search(query, facetFilters, start, count);
+        return SearchResultsMapper.map(searchResult, MlModelMapper::map);
+    }
+
+    @Override
+    public AutoCompleteResults autoComplete(@Nonnull String query,
+        @Nullable String field,
+        @Nullable List<FacetFilterInput> filters,
+        int limit,
+        @Nonnull final QueryContext context) throws Exception {
+        final Map<String, String> facetFilters = ResolverUtils.buildFacetFilters(filters, FACET_FIELDS);
+        field = field != null ? field : DEFAULT_AUTO_COMPLETE_FIELD;
+        final AutoCompleteResult result = _mlModelsClient.autoComplete(query, field, facetFilters, limit);
+        return AutoCompleteResultsMapper.map(result);
+    }
+
+    @Override
+    public BrowseResults browse(@Nonnull List<String> path,
+        @Nullable List<FacetFilterInput> filters,
+        int start,
+        int count,
+        @Nonnull final QueryContext context) throws Exception {
```

Comment:
Do you plan to implement browse in the client within this pr? If not I'd suggest making this a SearchableEntityType _only_, until that functionality gets in